okapi-core/src/main/java/org/folio/okapi/managers/TenantManager.java

```diff
@@ -1240,13 +1215,7 @@ private void loadTenants(Handler<ExtendedAsyncResult<Void>> fut) {
         fut.handle(new Failure<>(gres.getType(), gres.cause()));
       } else {
         Collection<String> keys = gres.result();
-        if (!keys.isEmpty()) {
-          logger.info("Not loading tenants, looks like someone already did");
-          fut.handle(new Success<>());
-        } else if (tenantStore == null) {
-          logger.info("No storage to load tenants from, so starting with empty");
-          fut.handle(new Success<>());
-        } else {
+        if (keys.isEmpty()) {
           loadTenants2(fut);
         }
```

Comment:
The else-clause with a fut.handle(...) is missing.

Consider using guard-clauses and merging laodTenants2 into loadTenants:

    tenants.getKeys(gres -> {
      if (gres.failed()) {
        fut.handle(new Failure<>(gres.getType(), gres.cause()));
        return;
      }
      Collection<String> keys = gres.result();
      if (! keys.isEmpty()) {
        logger.info("Not loading tenants, looks like someone already did");
        fut.handle(new Success<>());
        return;
      }
      tenantStore.listTenants(lres -> {
        if (lres.failed()) {
          fut.handle(new Failure<>(lres.getType(), lres.cause()));
          return;
        }
        CompList<List<Void>> futures = new CompList<>(ErrorType.INTERNAL);
        for (Tenant t : lres.result()) {
          Promise<Void> f = Promise.promise();
          tenants.add(t.getId(), t, f::handle);
          futures.add(f);
        }
        futures.all(fut);
      }
    }